internal/contentcache/contentcache.go

```diff
@@ -51,6 +53,38 @@ func (c *ContentCache) NewTempFile(rc io.ReadCloser) (gcsx.TempFile, error) {
 	return gcsx.NewTempFile(rc, c.tempDir, c.mtimeClock)
 }
 
+// AddOrReplace creates a new cache file or updates an existing cache file
+func (c *ContentCache) AddOrReplace(cacheObjectKey *CacheObjectKey, generation int64, rc io.ReadCloser) (gcsx.TempFile, error) {
+	if cacheObject, exists := c.fileMap[*cacheObjectKey]; exists {
+		cacheObject.Destroy()
+	}
+	metadata := &gcsx.TempFileObjectMetadata{
+		BucketName: cacheObjectKey.BucketName,
+		ObjectName: cacheObjectKey.ObjectName,
+		Generation: generation,
+	}
+	file, err := c.NewCacheFile(rc, metadata)
+	if err != nil {
+		return nil, fmt.Errorf("Could not AddOrReplace cache file: %w", err)
+	}
+	c.fileMap[*cacheObjectKey] = file
+	return file, err
+}
+
+// Get retrieves a file from the cache given the GCS object name and bucket name
+func (c *ContentCache) Get(cacheObjectKey *CacheObjectKey) (gcsx.TempFile, bool) {
+	file, exists := c.fileMap[*cacheObjectKey]
+	return file, exists
+}
+
+// Remove removes and destroys the specfied cache file and metadata on disk
+func (c *ContentCache) Remove(cacheObjectKey *CacheObjectKey) {
+	if cacheObject, exists := c.fileMap[*cacheObjectKey]; exists {
+		cacheObject.Destroy()
+	}
+	delete(c.fileMap, *cacheObjectKey)
```

Comment:
Why do we delete the entry even if the key does not exist?

Reply:
Delete should be a no-op if the key doesn't exist in the map, but I can move it inside.

Reply:
Please move it inside for better readability and less overhead.